src/components/ProfilePage.jsx

```diff
@@ -1,81 +1,147 @@
 import React from 'react';
+import { useState } from 'react';
 import swal from 'sweetalert2';
 import { Formik, Form, Field } from 'formik';
 import { updateUserInfo } from '../services';
 import { CustomInput } from '.';
-import {
-  submitButtonStyle,
-  loginFormContainerStyle,
-  profileCard
-} from '../styles';
+import { submitButtonStyle, loginFormContainerStyle } from '../styles';
 import { UserConsumer } from '../contexts';
 import { profilePageValidator } from '../helpers';
+import imageUrl from '../images/imageURL.jpg';
+import noImageUrl from '../images/noImageUrl.png';
 
 const ProfilePage = props => {
+  const [editStatus, setEditStatus] = useState(false);
+  const [editButtonName, seteditButtonName] = useState('Edit Profile');
+
+  const handleOnEdit = () => {
+    if (editButtonName === 'Edit Profile') {
+      seteditButtonName('Finished');
```

Comment:
Hmmm. The logic for the `editStatus` could be used for the edit button text here too actually so we don't have to have this `setEditButtonName`.

Do a ternary on the button text that says:

`isEditing ? Done : Edit`

And then change the `[editStatus, setEditStatus] to [isEditing, setIsEditing].` The reason I recommend this change of name is that you'd discover that most times, it is best to set variable names that have to do with actions to verbs (present or continuous or past. Whatever the case may be). Makes it clearer to the one who wants to maintain your code what is what.

let me know if anything here is unclear.